crates/smart-config/src/validation/mod.rs

```diff
@@ -50,6 +52,48 @@
 //!     }
 //! }
 //! ```
+//!
+//! ## Filtering
+//!
+//! Filtering reuses the `Validate` trait, but rather than failing, converts a value to `None`.
+//!
+//! ```
+//! use smart_config::validation;
+//! # use smart_config::{testing, DescribeConfig, DeserializeConfig, ErrorWithOrigin};
+//!
+//! #[derive(DescribeConfig, DeserializeConfig)]
+//! struct FilteringConfig {
+//!     /// Will convert `url: ''` to `None`.
+//!     #[config(filter(validation::NotEmpty))]
```

Comment:
`filter` feels a bit inappropriate here; my immediate interpretation was "NotEmpty values will be filtered _out_", e.g. only empty strings are allowed here.
Maybe `deserialize_if` would be a more descriptive name?